### PR TITLE
refactor: rework handler API

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

Rework handler API:
- Remove the generic bounds from `Runtime`, `{Module,Component}Instance` and everything in between
- Remove impls for `()`
- Introduce `Invocation` enums
- Handlers now return a singular level error, rather than "two-dimensional" ones
- `anyhow::Error` is used as a standard error type across the repository (could be changed in the future, but for now standardized Rust errors are still in flux)

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [x] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [x] aarch64-linux
- [x] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: src/capability/numbergen/mod.rs [hunk 1]
```diff
@@ -1,47 +1,56 @@
...
     /// Returns a random [u32]
-    ///
-    /// # Errors
-    ///
-    /// Returns [`Self::Error`] if the operation fails
-    async fn random_32(&self, claims: &jwt::Claims<jwt::Actor>) -> Result<u32, Self::Error>;
+    Random32,
+}
+
+impl<O: AsRef<str>, P: AsRef<[u8]>> TryFrom<(O, Option<P>)> for Invocation {
```
Comment (autodidaddict) [inline src/capability/numbergen/mod.rs:31]: this type signature makes my eyes water 🤣

**Review by autodidaddict — DISMISSED** [review]

File: src/capability/numbergen/mod.rs (hunk 1 above)
Comment (rvolosatovs) [reply src/capability/numbergen/mod.rs:31]: I'll move these to `where`

Comment (autodidaddict) [reply src/capability/numbergen/mod.rs:31]: i was mostly just kidding but yeah I think it would be easier to read as a where

Comment (rvolosatovs) [reply src/capability/numbergen/mod.rs]: updated

Comment (rvolosatovs) [reply src/capability/numbergen/mod.rs]: https://github.com/wasmCloud/wasmCloud/compare/169f3052cc579f854885a1be09ba89303c630fd7..3ddbe080c8e302d35019fbe5ca3c906c7186f61a

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**